### PR TITLE
Support ARM architecture

### PR DESCRIPTION
**Pull request by nemanja-tosic**
### Overview

Support ARM architecture.

### CODE REVIEW

**Comment by Ma27** [comment]
looks good so far, but it needs a test case before merging

**Comment by nemanja-tosic** [comment]
No problem, ill add a spec for it.

**Comment by Ma27** [comment]
ok, thank you. When you're done with it, I think that we can merge it :)

File: spec/defines/nodejs_instance_spec.rb [hunk 1]
```diff
@@ -196,4 +196,86 @@
...
+      :version              => 'v6.0.0',
+      :ensure               => 'present',
+      :make_install         => false,
+      :target_dir           => '/usr/local/bin',
+      :cpu_cores            => 1,
+      :default_node_version => 'v5.6.0',
+    }}
+
+    describe 'os' do
+     describe 'darwin' do
```
Comment (Ma27) [inline spec/defines/nodejs_instance_spec.rb]: the indention looks wrong here, please indent two chars

File: spec/defines/nodejs_instance_spec.rb [hunk 2]
```diff
@@ -196,4 +196,86 @@
...
+      let(:facts) {{
+        :hardwaremodel => '2.6.32-504.12.2.el6.x86_64',
+        :kernel        => 'darwin',
```
Comment (Ma27) [inline spec/defines/nodejs_instance_spec.rb]: please add `:processorcount => 1` as this fact isn't registered in the test case and it breaks the `strict vars` build on travis

File: spec/defines/nodejs_instance_spec.rb [hunk 3]
```diff
@@ -196,4 +196,86 @@
...
+
+      it { should contain_nodejs__instance__download('nodejs-download-v6.0.0') \
+        .with_source('https://nodejs.org/dist/v6.0.0/node-v6.0.0-darwin-x64.tar.gz') \
+      }
+     end
+
+     describe 'linux' do
+      let(:facts) {{
+        :hardwaremodel => '2.6.32-504.12.2.el6.x86_64',
+        :kernel        => 'linux',
```
Comment (Ma27) [inline spec/defines/nodejs_instance_spec.rb]: the `:processorcount` param ism issing here, too...

File: spec/defines/nodejs_instance_spec.rb [hunk 4]
```diff
@@ -196,4 +196,86 @@
...
+        .with_source('https://nodejs.org/dist/v6.0.0/node-v6.0.0-linux-x64.tar.gz') \
+      }
+     end
+    end
+
+    describe 'arch' do
+      describe 'x64' do
+        let(:facts) {{
+          :hardwaremodel => '2.6.32-504.12.2.el6.x86_64',
+          :kernel        => 'linux',
```
Comment (Ma27) [inline spec/defines/nodejs_instance_spec.rb]: same here...

File: spec/defines/nodejs_instance_spec.rb [hunk 5]
```diff
@@ -196,4 +196,86 @@
...
+        it { should contain_nodejs__instance__download('nodejs-download-v6.0.0') \
+          .with_source('https://nodejs.org/dist/v6.0.0/node-v6.0.0-linux-x64.tar.gz') \
+        }
+      end
+
+      describe 'ARM' do
+        describe 'armv6l' do
+          let(:facts) {{
+            :hardwaremodel => 'armv6l',
+            :kernel        => 'linux',
```
Comment (Ma27) [inline spec/defines/nodejs_instance_spec.rb]: and here...

File: spec/defines/nodejs_instance_spec.rb [hunk 6]
```diff
@@ -196,4 +196,86 @@
...
+          }}
+
+          it { should contain_nodejs__instance__download('nodejs-download-v6.0.0') \
+            .with_source('https://nodejs.org/dist/v6.0.0/node-v6.0.0-linux-armv6l.tar.gz') \
+          }
+        end
+        describe 'armv7l' do
+          let(:facts) {{
+            :hardwaremodel => 'armv7l',
+            :kernel        => 'linux',
```
Comment (Ma27) [inline spec/defines/nodejs_instance_spec.rb]: and here...

File: spec/defines/nodejs_instance_spec.rb [hunk 7]
```diff
@@ -196,4 +196,86 @@
...
+          it { should contain_nodejs__instance__download('nodejs-download-v6.0.0') \
+            .with_source('https://nodejs.org/dist/v6.0.0/node-v6.0.0-linux-armv7l.tar.gz') \
+          }
+        end
+      end
+
+      describe 'default' do
+        let(:facts) {{
+          :hardwaremodel => '2.6.32-504.12.2.el6.x86',
+          :kernel        => 'linux',
```
Comment (Ma27) [inline spec/defines/nodejs_instance_spec.rb]: and here...

**Comment by Ma27** [comment]
thanks @nemanja-tosic :tada:

**Pull request merged**